### PR TITLE
Explore: Execute Lucene metric queries through the backend 

### PR DESCRIPTION
**Pull request by fridgepoet**
**What this PR does / why we need it**:
In Grafana Explore, metric aggregations such as Count, Sum, Min, Max, Average, Extended Stats, Percentiles, and Unique Count (and pipeline aggregations) are now executed through the backend.

**Queries with these metric aggregations will only run in the backend in the Explore view** in order to limit this migration's impact to dashboards. Please [let us know](https://github.com/grafana/opensearch-datasource/issues/new?assignees=&labels=datasource%2FOpenSearch%2Ctype%2Fbug&projects=&template=bug_report.md) if you observe any unwanted behavior.

**Which issue(s) this PR fixes**:
Closes https://github.com/grafana/opensearch-datasource/issues/197

**Special notes for your reviewer**:
Differences between backend and frontend data frames:
- Backend data frames have additional `"typeInfo": {
              "frame": "float64",
              "nullable": true` This seems acceptable as this is how the equivalent Elasticsearch query also returns data frames.
- For a query with Alias, backend data frames have additional "config" object with "displayNameFromDS": "distanceKms". This is expected due to https://github.com/grafana/opensearch-datasource/pull/180. This is also the same behavior in Elasticsearch for the same type of query.
- Backend data frames have additional `"meta": {
          "type": "timeseries-multi",
          "typeVersion": [
            0,
            0
          ]
        }` object. This is due to code based on an Elasticsearch PR https://github.com/grafana/grafana/pull/60169 and was included in https://github.com/grafana/opensearch-datasource/pull/180 and https://github.com/grafana/opensearch-datasource/pull/168. This is also the same behavior in Elasticsearch for the same type of query.
- Backend has additional ` "labels": {
              "metricId": "1"
            }` object. The presence and use of this object is consistent with Elasticsearch for the same type of query.

[Full differences and actions taken available in the description of the issue](https://github.com/grafana/opensearch-datasource/issues/197)

### CODE REVIEW

**Comment by idastambuk** [comment]
Impressive investigation @fridgepoet! I clicked around running different types of queries and all looks great!

**Review by idastambuk — APPROVED** [review]

File: src/datasource.ts [hunk 1]
```diff
@@ -541,7 +541,7 @@ export class OpenSearchDatasource extends DataSourceWithBackend<OpenSearchQuery,
             metric =>
               metric.type === 'raw_data' ||
               metric.type === 'raw_document' ||
-              (request.app === CoreApp.Explore && target.queryType === QueryType.Lucene && metric.type === 'logs')
+              (request.app === CoreApp.Explore && target.queryType === QueryType.Lucene && metric.type)
```
Comment (kevinwcyu) [inline src/datasource.ts:544]:
```suggestion
              (request.app === CoreApp.Explore && target.queryType === QueryType.Lucene)
```

now that we are sending all lucene explore queries to the backend, do we need the check for metric.type here at all? would it ever be undefined, and if it is, would it matter? this is just a minor nit.

**Review by kevinwcyu — APPROVED** [review]
nice outline of the differences and explanations 🎉

File: src/datasource.ts (hunk 1 above)
Comment (fridgepoet) [reply src/datasource.ts]: Was wondering this myself, tried it out locally. Didn't see anything weird.
I'll merge with your suggestion and continue keeping an eye out. Thank yoou!

**Pull request merged**